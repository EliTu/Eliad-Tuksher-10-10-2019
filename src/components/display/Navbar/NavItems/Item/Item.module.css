@import '../../../../../variables.css';

.ItemStyles {
    display: flex;
    align-items: center;
    justify-content: center;
    font-size: var(--text-size-medium);
    margin: 0 auto;
    color: var(--color-main-grey-dark);
    border-bottom: var(--border-thick);
    border-color: rgba(255, 255, 255, 0);
    height: 100%;
    width: 50%;
    transition: all 0.3s ease-in;
}

.ItemStyles:hover {
    background-color: rgba(66, 159, 235, 0.918);
}

.active {
    border-bottom-color: var(--color-main-black);
}