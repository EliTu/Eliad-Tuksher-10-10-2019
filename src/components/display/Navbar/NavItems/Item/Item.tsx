import React from 'react';
import { NavLink } from 'react-router-dom';
import styles from './Item.module.css';

interface IProps {
	children: string;
	link: string;
}

const Item: React.FC<IProps> = ({ children, link }) => {
	const { ItemStyles, active } = styles;

	return (
		<NavLink
			className={ItemStyles}
			to={link}
			activeClassName={active}
			exact
		>
			{children}
		</NavLink>
	);
};

export default Item;
