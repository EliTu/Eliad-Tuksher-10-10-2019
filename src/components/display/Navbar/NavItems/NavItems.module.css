.NavItemsStyles {
    display: flex;
    align-items: flex-end;
    justify-content: flex-end;
    justify-items: flex-end;
    height: 100%;
    width: 25%;
}