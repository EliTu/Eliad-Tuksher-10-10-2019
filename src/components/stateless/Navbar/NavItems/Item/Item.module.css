@import '../../../../../variables.css';

.ItemStyles {
    font-size: var(--text-size-medium);
    margin: 2rem;
}