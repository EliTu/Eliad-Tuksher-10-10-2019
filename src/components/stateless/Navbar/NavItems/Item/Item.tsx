import React from 'react';
import styles from '.Item.module.css';

const Item: React.FC = () => {
    const { ItemStyles } = styles;
    
	return <div className={ItemStyles}></div>;
};

export default Item;
