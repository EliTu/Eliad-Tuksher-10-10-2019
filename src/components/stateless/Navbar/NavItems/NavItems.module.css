.NavItemsStyle {
    display: flex;
    align-items: flex-end;
    justify-content: flex-end;
    justify-items: flex-end;
}