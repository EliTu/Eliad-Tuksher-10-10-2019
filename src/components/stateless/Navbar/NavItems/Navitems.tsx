import React from 'react';
import styles from './NavItems.module.css';

const Navitems: React.FC = () => {
	const { NavItemsStyle } = styles;

	return <div className={NavItemsStyle}></div>;
};

export default Navitems;
