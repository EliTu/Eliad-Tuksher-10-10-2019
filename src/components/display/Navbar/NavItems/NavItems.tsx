import React from 'react';
import Item from './Item/Item';
import styles from './NavItems.module.css';

const Navitems: React.FC = () => {
	const { NavItemsStyles } = styles;

	return (
		<div className={NavItemsStyles}>
			<Item link={'/'}>Home</Item>
			<Item link={'/favorites'}>My Favorites</Item>
			<Item link={'/about'}>About</Item>
		</div>
	);
};

export default Navitems;
