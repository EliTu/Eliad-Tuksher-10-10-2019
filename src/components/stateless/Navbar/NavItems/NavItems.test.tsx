import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import NavItems from './NavItems';
import Item from './Item/Item';
import '../../../../setupTests';

describe('NavItems component', () => {
	let component: ShallowWrapper;

	beforeEach(() => {
		component = shallow(<NavItems />);
	});

	it('should render without errors', () => {
		expect(component).toMatchSnapshot();
		expect(component.length).toBe(1);
		expect(component.length).not.toBe(2);
		expect(component).toHaveClassName('NavItemsStyle');
	});

	it('should render a div with 3 Item components with the texts Home, Favorites and About', () => {
		const container = component.find('div');

		expect(container.length).toBe(1);
		expect(container.children()).toHaveLength(3);
		expect(container).toContainReact(<Item>Home</Item>);
		expect(container).toContainReact(<Item>About</Item>);
		expect(container).toContainReact(<Item>Favorites</Item>);
	});
});
