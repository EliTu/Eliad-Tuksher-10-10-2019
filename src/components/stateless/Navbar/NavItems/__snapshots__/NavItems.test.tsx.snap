// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavItems component should render without errors 1`] = `
<div
  className="NavItemsStyle"
>
  <Item>
    Home
  </Item>
  <Item>
    Favorites
  </Item>
  <Item>
    About
  </Item>
</div>
`;
